org-node/src/client.rs

```diff
@@ -352,23 +351,39 @@ impl Client {
             }
             Request::UpdateRefs(urn, rx) => {
                 tracing::info!(target: "org-node", "updating refs for urn: {:?}", urn);
+
                 // sign the updated ref;
-                api.using_storage({
-                    let urn = urn.clone();
-                    move |storage| match Refs::update(storage, &urn) {
-                        Ok(_u) => {
-                            tracing::info!(target: "org-node", "updated signed refs");
-                            // Respond with successful callback
-                            Ok(())
-                        }
-                        Err(e) => {
-                            tracing::error!(target: "org-node", "failed to update refs");
-                            Err(Error::from(Box::new(e)))
-                        }
-                    }
-                })
-                .await?
-                .map_err(Error::from)?;
+                Client::sign_refs(urn.clone(), api).await?;
+
+                let project = Client::get_project_head(&urn, api).await?;
+
+                // Return the project's default branch;
+                let Head { branch, .. } =
+                    project.ok_or_else(|| Error::Project(urn.clone(), "failed to find project"))?;
+
+                let namespace = urn.encode_id();
+
+                let namespace_path = Path::new("refs").join("namespaces").join(&namespace);
```

Comment:
Note: setting the repository namespace, i.e. `repository.set_namespace(...)` did not correctly set the symbolic reference, therefore we need to manually provide the namespace path, as is done in the `TrackProject` handle.